minimatch.js

```diff
@@ -647,9 +647,9 @@ class Minimatch {
       // this is a huge pita.  We now have to re-walk
       // the contents of the would-be class to re-translate
       // any characters that were passed through as-is
-      cs = pattern.substr(classStart + 1)
+      cs = pattern.slice(classStart + 1)
       sp = this.parse(cs, SUBPARSE)
-      re = re.substr(0, reClassStart) + '\\[' + sp[0]
+      re = re.substring(0, reClassStart) + '\\[' + sp[0]
```

Comment:
Why `substring` here, and not `slice`?

Reply:
`reClassStart` could be `-1`, in that case `substring()` will return an empty string like `substr()`, `slice()` however would return everything from the start until the second last char. This would be a different behaviour than we currently have.

```
'example'.slice(0, -1) == 'exampl'
'example'.substring(0, -1) == ''
'example'.substr(0, -1) == ''
```
I generally like `slice()` more [as it's stricter](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/substring#differences_between_substring_and_slice) and doesn't switch args around but in this case it's not the right tool to use to have the same result as `substr()`.

Reply:
In that case, maybe use `substring()` to have consistent results with existing `substr()` so that it's non-breaking?